tests/dualtor/test_tor_ecn.py

```diff
@@ -0,0 +1,344 @@
+):
+    """
+    Verify ECN value on the received packet w.r.t expected packet
+    """
+    _, rec_pkt = testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=[exp_ptf_port_index])
+    rec_pkt = Ether(rec_pkt)
+    logging.info("received packet:\n%s", dump_scapy_packet_show_output(rec_pkt))
+
+    rec_dscp = rec_pkt[IP].tos >> 2
+    rec_ecn = rec_pkt[IP].tos & 3
+
+    if rec_ecn != exp_ecn:
+        pytest.fail("the expected ECN: {0:02b} not matching with received ECN: {0:02b}".format(exp_ecn, rec_ecn))
+    else:
+        logging.info("the expected ECN: {0:02b} matching with received ECN: {0:02b}".format(exp_ecn, rec_ecn))
+
+def test_dscp_to_queue_during_decap_on_active(
+    apply_active_state_to_orchagent,
+    build_encapsulated_ip_packet,
+    rand_selected_interface, 
+    ptfadapter,
+    tbinfo, 
+    rand_selected_dut, 
+    tunnel_traffic_monitor, 
+    duthosts, 
+    rand_one_dut_hostname
+):
+    """
+    Test if DSCP to Q mapping for inner header is matching with outer header during decap on active
+    """
+    tor = rand_selected_dut
+    encapsulated_packet = build_encapsulated_ip_packet
+    iface, _ = rand_selected_interface
+
+    exp_ptf_port_index = get_ptf_server_intf_index(tor, tbinfo, iface)
+    exp_pkt = build_expected_packet_to_server(encapsulated_packet)
+
+    # Clear queue counters
+    duthost = duthosts[rand_one_dut_hostname]
+    duthost.shell('sonic-clear queuecounters')
+    logging.info("Clearing queue counters before starting traffic")
+
+    ptfadapter.dataplane.flush()
+    ptf_t1_intf = random.choice(get_t1_ptf_ports(tor, tbinfo))
+    logging.info("send encapsulated packet from ptf t1 interface %s", ptf_t1_intf)
+    testutils.send(ptfadapter, int(ptf_t1_intf.strip("eth")), encapsulated_packet, count=10)
+
+    exp_tos = encapsulated_packet[IP].payload[IP].tos
+    exp_dscp = exp_tos >> 2
+    exp_queue = derive_queue_id_from_dscp(exp_dscp)
+
+    _, rec_pkt = testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=[exp_ptf_port_index])
+    rec_pkt = Ether(rec_pkt)
+    logging.info("received decap packet:\n%s", dump_scapy_packet_show_output(rec_pkt))
+
+    time.sleep(10)
+    rec_queue = get_queue_id_of_received_packet(duthosts, rand_one_dut_hostname, rand_selected_interface)
+
+    if rec_queue == None or rec_queue != exp_queue:
+        pytest.fail("the expected Queue : {} not matching with received Queue : {}".format(exp_queue, rec_queue))
+    else:
+        logging.info("the expected Queue : {} matching with received Queue : {}".format(exp_queue, rec_queue))
+
+def test_dscp_to_queue_during_encap_on_standby(
+    build_non_encapsulated_ip_packet,
+    rand_selected_interface, ptfadapter,
+    tbinfo, 
+    rand_selected_dut, 
+    tunnel_traffic_monitor, 
+    duthosts, 
+    rand_one_dut_hostname
+):
+    """
+    Test if DSCP to Q mapping for outer header is matching with inner header during encap on standby
+    """
+    rand_selected_dut.shell("/usr/local/bin/write_standby.py")
```

Comment:
@theasianpianist , pre-requisite for this is the presence of write_standby.py file. How shall we make it available ?

Reply:
This file is present in internal images. As long as the `apply_peer_switch_table_to_dut` fixture is called it should behave as expected.

Reply:
@theasianpianist could you please attach write_standby.py to this PR? TC fails as can not find it:
`/bin/sh: 1: /usr/local/bin/write_standby.py: not found"`

SONiC version:
`SONiC Software Version: SONiC.master.23312-dirty-20210710.090215
Distribution: Debian 10.10
Kernel: 4.19.0-12-2-amd64
Build commit: 447f09f2d
Build date: Sat Jul 10 14:40:36 UTC 2021
Built by: AzDevOps@sonic-build-workers-000H8R

Platform: x86_64-arista_7170_64c
HwSKU: Arista-7170-64C
ASIC: barefoot
ASIC Count: 1
Model Number: DCS-7170-64C`